### PR TITLE
Add option to force fully qualified names in PDSC encoder.

### PR DESCRIPTION
**Pull request by karthikbalasub**

### CODE REVIEW

File: data/src/main/java/com/linkedin/data/schema/SchemaToJsonEncoder.java [hunk 1]
```diff
@@ -95,6 +95,7 @@ public static String schemasToJson(Collection<DataSchema> schemas, JsonBuilder.P
   protected final JsonBuilder _builder;
   protected String _currentNamespace = "";
   protected String _currentPackage = "";
+  protected boolean _alwaysUseFullyQualifedName = false;
```
Comment (aman1309) [inline data/src/main/java/com/linkedin/data/schema/SchemaToJsonEncoder.java:98]: private?

File: data/src/main/java/com/linkedin/data/schema/SchemaToJsonEncoder.java [hunk 2]
```diff
@@ -127,6 +128,14 @@ public String getCurrentNamespace()
     return _currentNamespace;
   }
 
+  /**
+   * Configure the encoder to always use fully qualified names when encoding named type references.
```
Comment (aman1309) [inline data/src/main/java/com/linkedin/data/schema/SchemaToJsonEncoder.java:132]: can you add when this should be set

File: restli-tools/src/main/java/com/linkedin/restli/tools/data/SchemaFormatTranslator.java [hunk 3]
```diff
@@ -90,6 +89,10 @@
     OPTIONS.addOption(OptionBuilder.withLongOpt("skip-verification")
         .withDescription("Skip verification of the translated schemas. Be cautious, as incorrect translations will not be caught.")
         .create('k'));
+
+    OPTIONS.addOption(OptionBuilder.withLongOpt("force-pdsc-fully-qualified-names")
+        .withDescription("Forces generated PDSC schemas to always use fully qualified names.")
```
Comment (aman1309) [inline restli-tools/src/main/java/com/linkedin/restli/tools/data/SchemaFormatTranslator.java:94]: same here

File: data/src/main/java/com/linkedin/data/schema/SchemaToJsonEncoder.java (hunk 2 above)
Comment (nickibi) [reply data/src/main/java/com/linkedin/data/schema/SchemaToJsonEncoder.java:132]: This is for LinkedIn specific use case, we should avoid adding those information in the open source code base.

Comment (karthikbalasub) [reply data/src/main/java/com/linkedin/data/schema/SchemaToJsonEncoder.java:132]: Updated the javadoc to provide more details.

**Pull request merged**